### PR TITLE
fix(material/schematics): gracefully skip broken css files

### PR DESCRIPTION
**Pull request by mmalerba**
also don't treat files like something.css.map as a CSS file.

### CODE REVIEW

File: src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts [hunk 1]
```diff
@@ -37,7 +37,12 @@ export class ThemingStylesMigration extends Migration<ComponentMigrator[], Schem
       },
     ]);
 
-    return processor.process(styles, {syntax: scss}).toString();
+    try {
+      return processor.process(styles, {syntax: scss}).toString();
+    } catch (e) {
+      console.warn(`Failed to process stylesheet: `, filename);
```
Comment (crisbeto) [inline src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts:43]: Should we log the error here too?

File: src/material/schematics/ng-update/migrations/legacy-components-v15/index.ts [hunk 2]
```diff
@@ -37,7 +37,11 @@ export class LegacyComponentsMigration extends Migration<null> {
         RootExit: root => this._handleRootNode(root, stylesheet.filePath, namespace),
       },
     ]);
-    processor.process(stylesheet.content, {syntax: scss}).sync();
+    try {
+      processor.process(stylesheet.content, {syntax: scss}).sync();
+    } catch (e) {
```
Comment (crisbeto) [inline src/material/schematics/ng-update/migrations/legacy-components-v15/index.ts:42]: Also maybe log the error here?

**Review by crisbeto — APPROVED** [review]
LGTM

**Review by wagnermaciel — APPROVED** [review]
LGTM!

File: src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts [hunk 3]
```diff
@@ -37,7 +37,13 @@ export class ThemingStylesMigration extends Migration<ComponentMigrator[], Schem
       },
     ]);
 
-    return processor.process(styles, {syntax: scss}).toString();
+    try {
+      return processor.process(styles, {syntax: scss}).toString();
+    } catch (e) {
+      console.error(e);
+      console.warn(`Failed to process stylesheet: ${filename} (see error above).`);
```
Comment (devversion) [inline src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts:44]: Ideally you would use the migration logger that integrates with the CLI

**Review by devversion — APPROVED** [review]
Nice, good fix with the regex. Interesting that maps are part of the source tree at the same time though

**Comment by mmalerba** [comment]
Yeah people shouldn't have broken css or map files in the source tree I guess, but just making the script a little more robust so it doesn't blow up when it encounters those kinds of things.

**Pull request merged**